symbulate/results.py

```diff
@@ -495,50 +504,50 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
                 cbar.set_label('Relative Frequency')
                 caxes.yaxis.set_label_position("left")
             elif 'mixed-tile' in type:
+                oxmax, oxmin = max(x), min(x)
+                oymax, oymin = max(y), min(y)
                 if not discrete_x:
-                    x_bin = np.linspace(np.amin(x), np.amax(x), bins)
-                    x_new = []
-                    for val in x:
-                        i = 0
-                        while val > x_bin[i]:
-                            i += 1
-                        x_new.append(x_bin[i])
-                    x = x_new
-                    counts = count_var(list(zip(x, y)))
-                    x_uniq = np.unique(x)
-                    y_uniq = np.unique(y)
-                    xmin = min(x)
-                    ymin = min(y)
-                    xmax = max(x)
-                    ymax = max(y)
-                    x_pos = np.linspace(0, int(round(xmax)), bins)
-                    y_pos = np.arange((ymax + 1) - ymin)
-                    nums = len(x)
-
-                    intensity = np.zeros(shape=(len(y_pos), len(x_pos)))
+                    x_bin = np.linspace(min(x), max(x), bins)
+                    x = np.digitize(x, x_bin)
                 elif not discrete_y:
-                    y_bin = np.linspace(np.amin(y), np.amax(y), bins)
-                    y_new = []
-                    for val in y:
-                        i = 0
-                        while val > y_bin[i]:
-                            i += 1
-                        y_new.append(y_bin[i])
-                    y = y_new
-                    counts = count_var(list(zip(x, y)))
-                    x_uniq = np.unique(x)
-                    y_uniq = np.unique(y)
-                    xmin = min(x)
-                    ymin = min(y)
-                    xmax = max(x)
-                    ymax = max(y)
-                    x_pos = np.arange((xmax + 1) - xmin)
-                    y_pos = np.linspace(0, int(round(ymax)), bins)
-                    nums = len(x)
-
-                    intensity = np.zeros(shape=(len(y_pos), len(x_pos)))
-                print(intensity)
-                print(counts)
+                    y_bin = np.linspace(min(y), max(y), bins)
+                    y = np.digitize(y, y_bin)
+                x_uniq = np.unique(x)
+                y_uniq = np.unique(y)
+                xmax, xmin = max(x), min(x)
+                ymax, ymin = max(y), min(y)
+                x_pos = list(range(len(x_uniq)))
+                y_pos = list(range(len(y_uniq)))
+                x_map = dict(zip(x_uniq, x_pos))
```

Comment:
I think this is wrong. This seems to assume that x_uniq is sorted?

Reply:
Nvm. `np.unique()` returns the sorted unique values. This is worth a comment above.